features/step_definitions/borrowers_view_and_sign_deed.rb

```diff
@@ -0,0 +1,57 @@
+Then(/^borrower <(\d+)> views the deed$/) do |borrower|
+  # Use ONLY for the very first viewing of the deed
+  step %(I navigate to the borrower frontend "/borrower-reference" page)
+  step %(the deed id is returned by the Deed API)
+  step %(I retrieve the unique user id for borrower <#{borrower}>)
+  step %(I search for the deed using the unique borrower reference)
+  step %(I enter the date of birth for borrower <#{borrower}>)
+  step %(when I click on the "Continue" link)
+  step %(the mortgage deed is displayed)
+end
```

Comment:
I would make the steps for getting the deed ID part of the setup of the data 'super step' only because you only need to get the ID once, whereas the implementation above gets it every time you view the deed, which is sometimes four or five times in the scenarios above.